### PR TITLE
Fix ci-sonarqube-ts input format

### PR DESCRIPTION
**Pull request by manjunathb4461**
- fix input format by using `${{ }}` everywhere https://github.com/smartcontractkit/.github/pull/472#discussion_r1662637659
- I hadn't noticed this in my previous [PR](https://github.com/smartcontractkit/.github/pull/472) that [test artifacts generated](https://github.com/smartcontractkit/functions-toolkit/actions/runs/9696301884/job/26757939413) weren't downloaded properly because they were in a different workflow. Added support for downloading artifacts from different workflows.
- Tested `ci-sonarqube-ts` in [ci-sonarqube](https://github.com/smartcontractkit/functions-toolkit/actions/runs/9764321753/job/26952499524)
- Tested `ci-sonarqube-go` in [ci-sonarqube](https://github.com/smartcontractkit/chainlink-data-streams/actions/runs/9778712507/job/26996194231)

### CODE REVIEW

File: actions/ci-sonarqube-ts/action.yml [hunk 1]
```diff
@@ -71,28 +81,27 @@ runs:
       with:
         fetch-depth: ${{ inputs.checkout-repo-fetch-depth }}
 
-    - name: Download all workflow run artifacts
-      uses: actions/download-artifact@65a9edc5881444af0b9093a5e628f2fe47ea3b2e # v4.1.7
```
Comment (manjunathb4461) [inline actions/ci-sonarqube-ts/action.yml:75]: This only supports downloading artifacts only using workflow run id, which is hard to determine when the test and lint steps are run in different workflows

File: actions/ci-sonarqube-ts/action.yml [hunk 2]
```diff
@@ -71,28 +81,27 @@ runs:
...
+    - name: Download test-report artifact
+      uses: dawidd6/action-download-artifact@bf251b5aa9c2f7eeb574a96ee720e24f801b7c11 # v6
+      with:
+        workflow: ${{ inputs.test-report-workflow }}
+        name: ${{ inputs.test-report-artifact }}
 
-    - name: Set SonarQube Report Paths
```
Comment (manjunathb4461) [inline actions/ci-sonarqube-ts/action.yml:77]: we don't need this anymore as we're downloading all reports to cwd

File: actions/ci-sonarqube-ts/action.yml (hunk 1 above)
Comment (manjunathb4461) [reply actions/ci-sonarqube-ts/action.yml:75]: x-repos? This doesn't require access to other repos. The artifacts downloads will be from different workflows within the same repo.

Comment (momentmaker) [reply actions/ci-sonarqube-ts/action.yml:75]: ahh got it. that's cool. nice!

is there anyway to test this action? maybe create a fake ts dir in `releng-test`?

Comment (manjunathb4461) [reply actions/ci-sonarqube-ts/action.yml:75]: I tested this in [functions-toolkit](https://github.com/smartcontractkit/functions-toolkit/actions/runs/9764321753/job/26952499524) repo, looks good

**Review by momentmaker — APPROVED** [review]

**Comment by momentmaker** [comment]
ooh I forgot - could you add a changeset?

**Comment by manjunathb4461** [comment]
> ooh I forgot - could you add a changeset?

I forget it everytime, my bad. I've added the changeset now, PTAL

File: actions/ci-sonarqube-ts/action.yml (hunk 1 above)
Comment (manjunathb4461) [reply actions/ci-sonarqube-ts/action.yml:75]: Should we also make similar changes to ts-sonarqube-go action as well?

Comment (momentmaker) [reply actions/ci-sonarqube-ts/action.yml:75]: yeah let's do that too :)

Comment (manjunathb4461) [reply actions/ci-sonarqube-ts/action.yml:75]: Made the changes for `ci-sonarqube-go` and tested in [ci-sonarqube](https://github.com/smartcontractkit/chainlink-data-streams/actions/runs/9778712507/job/26996194231) step

Comment (momentmaker) [reply actions/ci-sonarqube-ts/action.yml:75]: cool nice!

**Review by njegosrailic — APPROVED** [review]

**Pull request merged**